web/app/js/components/Navigation.jsx

```diff
@@ -356,7 +368,12 @@ class NavigationBase extends React.Component {
 
   render() {
     const { api, classes, selectedNamespace, ChildComponent, ...otherProps } = this.props;
-    const { namespaces, anchorEl, showNamespaceChangeDialog, newNamespace, mobileSidebarOpen } = this.state;
+    let { namespaces, namespaceFilter, namespaceFilterInput, anchorEl, showNamespaceChangeDialog, newNamespace, mobileSidebarOpen } = this.state;
+    if (namespaceFilter !== '') {
```

Comment:
With my proposed change to the filter regex, you wouldn't need this if statement anymore.

Reply:
Good call!